### PR TITLE
PLTF-1140 add new table action “Scan Product Barcode”

### PR DESCRIPTION
**Pull request by volodymyrboichuk**
PLTF-1142 add new TA for stock dynamics table: “Change Stock”

### CODE REVIEW

File: src/api/Communicator.ts [hunk 1]
```diff
@@ -193,6 +192,92 @@ export class Communicator extends BaseCommunicator {
...
+			return this.fetchData(`api/products/insertProduct`, { warehouseId: activeWarehouseId }, product, {
+				method: 'POST',
+				ignoreTokens: false,
+				contentType: 'application/json',
+			});
+		});
+	}
+
+	public updateProductEanCode(productId: number, eanCode: string) {
+		console.log('productId', productId);
```
Comment (SerediukYar) [inline src/api/Communicator.ts]: log?

File: src/screens/CreateProductScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import KeyboardAvoidingWrapper from '../components/KeyboardAvoidingWrapper';
+// @ts-ignore
+import useAppContext from '../context/AppContext';
+import { Formik } from 'formik';
+import Communicator from '../api/Communicator';
+// @ts-ignore
+import { Toast } from 'popup-ui';
+
+const valueValidation = (value: number, charsBeforeComma: number): boolean => {
+	console.log('value', value);
```
Comment (SerediukYar) [inline src/screens/CreateProductScreen.tsx]: log?

File: src/components/InfiniteDropDown.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+interface IInfiniteDropDownProps {
+	defaultOpen?: boolean;
+	defaultValue?: string;
+	setValue: Function;
+}
+
+interface IProduct {
+	label: string;
+	value: number;
+
```
Comment (dakh-viktoriia) [inline src/components/InfiniteDropDown.tsx:23]: extra line

File: src/components/BarcodeScanner.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+	const handleScanning = useCallback((scannedInfo) => {
+		setScanned(true);
+		handleBarCodeScanned(scannedInfo);
+	}, []);
+	if (hasPermission === null) {
+		return (
+			<View style={styles.container}>
+				<TouchableOpacity onPress={requestPermission}>
+					<Text>Requesting for camera permission</Text>
```
Comment (SerediukYar) [inline src/components/BarcodeScanner.tsx:43]: message key

File: src/screens/CreateStockRecordScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+// @ts-ignore
+const CreateStockRecordScreen = ({ navigation, route }) => {
+	const { t, mappedRacksById } = useAppContext();
+	const [open, setOpen] = useState<boolean>(false);
+	const productId = useMemo(() => route.params.productId, [route.params]);
+
+	// const [multiplier, setMultiplier] = useState(1);
+	// const [isQuantityValid, setQuantityValid] = useState('');
+	//
```
Comment (dakh-viktoriia) [inline src/screens/CreateStockRecordScreen.tsx]: ?????????

File: src/screens/CreateStockRecordScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+	}, [mappedRacksById]);
+
+	// const newQuantityValue = useMemo(() => +quantityValue * multiplier, [multiplier, quantityValue]);
+	// const checkValidation = useCallback((quantity: number) => {
+	// 	const isValid = (() => {
+	// 		if (multiplier === 1) return Math.abs(+quantityValue + quantity) > JAVA_INT_MAX_VALUE;
+	// 		if (multiplier === -1) return +quantityValue > quantity;
+	// 	})();
+	// 	setQuantityValid(isValid);
+	// }, [multiplier]);
```
Comment (dakh-viktoriia) [inline src/screens/CreateStockRecordScreen.tsx]: ?????????

File: src/screens/Login.tsx [hunk 7]
```diff
@@ -51,7 +51,7 @@ const Login = ({ navigation }) => {
 					<SubTitle>{t('LOG_IN_TO_CONTINUE')}</SubTitle>
 
 					<Formik
-						initialValues={{ email: '', password: '' }}
+						initialValues={{ email: 'admin', password: '23mOHi,' }}
```
Comment (dakh-viktoriia) [inline src/screens/Login.tsx]: securiry? no, we didint hear

**Review by dakh-viktoriia — CHANGES_REQUESTED** [review]

**Review by dakh-viktoriia — APPROVED** [review]

**Review by SerediukYar — APPROVED** [review]

**Pull request merged**